### ISSUE TITLE
Pre-specifying lambda grid does not seem to work

### ISSUE THREAD

**Issue by tomwenseleers** [hazimehh/L0Learn#42]
With both the CRAN and github version L0Learn.fit and L0Learn.cvfit if I specify autoLambda=FALSE and lambdaGrid = as.list(lambdas) with a custom vector lambdas I get a fit back in which just the first lambda value is used. Would you happen to have  a working example of using a custom lambda grid?

**Comment by rahulmaz** [comment]
You may consider checking (the L0Learn Vignette for user specified tuning-parameter grids
http://www.mit.edu/~hazimeh/L0Learn-vignette.html#user-specified_lambda_grids
for a working example on how to specify tuning-parameter grids.

I have not seen your particular example, but you may consider playing around with 
the "maxSuppSize" argument --- if this is too small (compared the tuning parameter grid you have supplied) then L0Learn will exit with fewer solutions.

**Comment by tomwenseleers** [comment]
Thanks for this - so here is a reproducible example:
```
library(devtools)
install_github("hazimehh/L0Learn")
library(L0Learn)
data <- GenSynthetic(n=200,p=400,k=10,seed=1)

# first lets try with automatic lambda selection:
fit <- L0Learn.cvfit(x = data$X, y = data$y, loss = "SquaredError", penalty = "L0",
                              algorithm = "CDPSI", maxSuppSize = 100, nLambda = 100, nGamma = 0,
                              activeSet = TRUE,
                              autoLambda = TRUE,
                              # lambdaGrid = as.list(rev(lambdas)),
                              intercept = FALSE,
                              nFolds = 10, seed = 1)
coef(fit)
dim(coef(fit)) # 400 x 40 lambda values that were retained (note I asked nLambda=100 though)
fit$fit$lambda[[1]]   
```

Now imagine we would specify the lambda values returned by the fit above as pre-specified ```
lambdas:

lambdas <- fit$fit$lambda[[1]] # pre-specified lambda grid, here we take the same as used above
fit_lambdas <- L0Learn.cvfit(x = data$X, y = data$y, loss = "SquaredError", penalty = "L0",
                     algorithm = "CDPSI", maxSuppSize = 100, nLambda = length(lambdas), nGamma = 0,
                     activeSet = TRUE,
                     autoLambda = FALSE,
                     lambdaGrid = as.list(lambdas),
                     intercept = FALSE,
                     nFolds = 10, seed = 1)
coef(fit_lambdas)
dim(coef(fit_lambdas)) # 400 x 1 - only a single lambda value retained...
fit_lambdas$fit$lambda[[1]]   
```

We then just get coefficients for a single lambda back, which doesn't seem right... 

Or am I doing something wrong?

lambdas <- fit$fit$lambda[[1]] # pre-specified lambda grid, here we take the same as used above
fit_lambdas <- L0Learn.cvfit(x = data$X, y = data$y, loss = "SquaredError", penalty = "L0",
                     algorithm = "CDPSI", maxSuppSize = 100, nLambda = length(lambdas), nGamma = 0,
                     activeSet = TRUE,
                     autoLambda = FALSE,
                     lambdaGrid = as.list(lambdas),
                     intercept = FALSE,
                     nFolds = 10, seed = 1)
coef(fit_lambdas)
dim(coef(fit_lambdas)) # 400 x 1 - only a single lambda value retained...
fit_lambdas$fit$lambda[[1]]

**Comment by hazimehh** [comment]
The lambdaGrid parameter should be supplied a list, whose first element is a sequence of lambdas. Your example has a list of lambdas (which is causing the problem). A working example is below:

```R
userLambda <- list()
userLambda[[1]] <- lambdas

fit_lambdas <- L0Learn.cvfit(x = data$X, y = data$y, loss = "SquaredError", penalty = "L0",
                     algorithm = "CDPSI", maxSuppSize = 100, nLambda = length(lambdas), nGamma = 0,
                     activeSet = TRUE,
                     autoLambda = FALSE,
                     lambdaGrid = userLambda,
                     intercept = FALSE,
                     nFolds = 10, seed = 1)
coef(fit_lambdas)
dim(coef(fit_lambdas))
fit_lambdas$fit$lambda[[1]]
```

**Comment by hazimehh** [comment]
For the first code snipped, increasing maxSuppSize (for example = 400) returns back the 100 solutions

**Comment by tomwenseleers** [comment]
Many thanks - much appreciated!
